packages/webapp/src/routes/land/correct-email.js

```diff
@@ -0,0 +1,43 @@
+import constants from '../../utils/constants.js'
+import { validateEmail } from '../../utils/helpers.js'
+
+const handlers = {
+  get: async (_request, h) => {
+    const emailAddress = _request.yar.get(constants.redisKeys.LAND_OWNER_EMAIL)
+    return h.view(constants.views.CORRECT_OWNER_EMAIL, {
+      emailAddress
+    })
+  },
+  post: async (request, h) => {
+    let vewPage
+    if (request.payload.correctEmail === 'yes') {
+      setEmailSetails(request)
+      vewPage = h.redirect(constants.routes.CHECK_YOUR_DETAILS)
+    } else {
+      const emailStatus = validateEmail(request.payload.emailAddress)
+      if (!emailStatus) {
+        setEmailSetails(request)
+        vewPage = h.redirect(constants.routes.CHECK_YOUR_DETAILS)
```

Comment:
I think on `/correct-email` page you need to check for referer and redirect if set.